File: December 2023/How Many X s/main.cpp
// Link: https://www.geeksforgeeks.org/problems/how-many-xs4514/1

#include <bits/stdc++.h>
using namespace std;

// User function Template for C++

class Solution
{
public:
    int countX(int L, int R, int X)
    {
        // code here
        int ans = 0;
        while (++L < R)
        {
            int val = L;
            while (val)
            {
                if (val % 10 == X)
                    ans++;
                val /= 10;
            }
        }
        return ans;
    }
};

int main()
{
    Solution obj;
    int L = 10;
    int R = 19;
    int X = 1;
    cout << obj.countX(L, R, X) << endl;
    return 0;
}

File: December 2023/How Many X s/main.py
# Link: https://www.geeksforgeeks.org/problems/how-many-xs4514/1

# User function Template for python3


class Solution:
    def count1(self, n, X):
        count = 0
        while n > 0:
            if n % 10 == X:
                count += 1
            n = n // 10
        return count

    def countX(self, L, R, X):
        # code here
        count = 0
        for i in range(L + 1, R):
            count += self.count1(i, X)
        return count


if __name__ == "__main__":
    obj = Solution()
    L = 10
    R = 19
    X = 1
    print(obj.countX(L, R, X))
